File: 202-happy-number/happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        v=set()
        def pow(n):
            z=0
            while n:
                y=n%10
                z+=y**2
                n//=10
            return z
        while n not in v:
            v.add(n)
            n=pow(n)
            if n==1:
                return True
        return False

File: 202-happy-number/happy-number.java
class Solution {
    public boolean isHappy(int n) {
        int fast=n;
        int slow=n;
        do{
            slow=findSquare(slow);
            fast=findSquare(findSquare(fast));
            if(slow==1){
                return true;
            }
        }while(slow!=fast);

        return false;
    }
    private int findSquare(int n){
        int val=0;
        while(n>0){
            int r=n%10;
            val+=r*r;
            n/=10;
        }
        return val;
    }
}